_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,637 @@
+  TYPE                 Regular Web Application
+  CALLBACKS            ***
+  ALLOWED LOGOUT URLS  ***
+  ALLOWED ORIGINS
+  ALLOWED WEB ORIGINS
+  TOKEN ENDPOINT AUTH
+  GRANTS               implicit, authorization_code, refresh_token, client_credentials
+
+ ▸    Quickstarts: https://auth0.com/docs/quickstart/webapp
+ ▸    Hint: Emulate this app's login flow by running `auth0 test login ***`
+ ▸    Hint: Consider running `auth0 quickstarts download ***`
+```
+
+**NOTE**: The client secret is [not displayed](https://github.com/auth0/auth0-cli/issues/488) in the CLI output for regular applications. You can run `auth0 apps open` to open a browser in the application configuration. Then copy the secret from the **Settings** tab. In the example above, the tenant is `dev-avup2laz.us.auth0.com`.
+
+
+### Create users and roles
+
+As Auth0 will be used as the identity provider, you must create some test users. With auth0-cli:
+
+```shell
+auth0 users create
+```
+
+Complete the required information, you will see an output like the lines below:
+
+```text
+Connection Name: Username-Password-Authentication
+ Name: Patrick
+ Email: patrick@email.com
+ Password: ********
+
+=== dev-avup2laz.us.auth0.com user created
+
+  ID          auth0|643ec0e1e671c7c9c5916ed6    
+  EMAIL       patrick@email.com                 
+  CONNECTION  Username-Password-Authentication  
+```
+
+Save the ID for later.
+
+For this example, Auth0 will also be used for role management. JHipster applications are generated with authorization based on two roles that can be assigned to the user: `ROLE_USER` and `ROLE_ADMIN`. Create those roles in Auth0:
+
+```shell
+autho0 roles create
+```
+
+You will see an output like the following:
+
+```text
+Name: ROLE_USER
+Description: A user
+
+=== dev-avup2laz.us.auth0.com role created
+
+ ID           rol_175cvyWy20sxohgo  
+ NAME         ROLE_USER             
+ DESCRIPTION  A user     
+```
+
+Assign the roles to the users you created:
+
+```shell
+auth0 users roles assign
+```
+
+Follow the steps, you will see the output below:
+
+```text
+User ID: auth0|643ec0e1e671c7c9c5916ed6
+? Roles rol_24d61Zxpvuas66tF (Name: ROLE_ADMIN), rol_175cvyWy20sxohgo (Name: ROLE_USER)
+
+=== dev-avup2laz.us.auth0.com roles assigned to user (2)
+
+  ID                    NAME        DESCRIPTION       
+  rol_24d61Zxpvuas66tF  ROLE_ADMIN  An administrator  
+  rol_175cvyWy20sxohgo  ROLE_USER   A user            
+```
+
+### Configure a Login Action
+
+Besides the roles assignment just explained, the authentication flow must be customized to add the roles and the username to the custom token claims expected by JHipster applications. The way to accomplish this task with Auth0 is by adding a Login Action.
+
+First [configure the editor](https://github.com/auth0/auth0-cli#customization) to use in the environment:
+
+```shell
+export EDITOR=nano
+```
+
+Then create the Login Action:
+
+```shell
+auth0 actions create
+```
```

Comment:
This is soooo cool! I didn't know you could create actions with the Auth0 CLI. 🤩